### PR TITLE
MASTERDIET-6 - Logging in security

### PR DESCRIPTION
**Pull request by jakubowiczish**

### CODE REVIEW

File: src/main/java/pl/agh/edu/master_diet/security/TokenAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getJwtFromRequest(HttpServletRequest request) {
+        final String bearerToken = request.getHeader("Authorization");
+
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/security/TokenAuthenticationFilter.java]: Fixed

File: src/main/java/pl/agh/edu/master_diet/repository/DbInfoRepository.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Date;
+
+@Slf4j
+@Repository
+public class DbInfoRepository
+        extends MasterDietPersistence implements InfoRepository {
+
+    @Override
+    public ZonedDateTime getSysdate() {
+        log.info("Próba pobrania daty systemowej bazy danych");
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/repository/DbInfoRepository.java]: Fixed

File: src/main/java/pl/agh/edu/master_diet/config/log/RequestAndResponseLoggingFilter.java [hunk 3]
```diff
@@ -33,45 +32,49 @@
     );
 
     private static void logRequestHeader(ContentCachingRequestWrapper request, String prefix) {
-        val queryString = request.getQueryString();
+        var queryString = request.getQueryString();
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/config/log/RequestAndResponseLoggingFilter.java]: Fixed

File: src/main/java/pl/agh/edu/master_diet/config/WebMvcConfig.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package pl.agh.edu.master_diet.config;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.CorsRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+@Configuration
+public class WebMvcConfig implements WebMvcConfigurer {
+
+    private final long MAX_AGE_SECS = 3600;
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/config/WebMvcConfig.java:10]: I reckon that this final variable should not be put in application.properties as there is no need for that right now and probably never will

File: src/main/java/pl/agh/edu/master_diet/config/SecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                .redirectionEndpoint()
+                .baseUri("/oauth2/callback/*")
+                .and()
+                .userInfoEndpoint()
+                .userService(customOAuth2UserService)
+                .and()
+                .successHandler(oAuth2AuthenticationSuccessHandler)
+                .failureHandler(oAuth2AuthenticationFailureHandler);
+
+        // Add our custom Token based authentication filter
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/config/SecurityConfig.java]: Cleaned up

File: src/main/java/pl/agh/edu/master_diet/config/SecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+    @Autowired
+    private HttpCookieOAuth2AuthorizationRequestRepository httpCookieOAuth2AuthorizationRequestRepository;
+
+    @Bean
+    public TokenAuthenticationFilter tokenAuthenticationFilter() {
+        return new TokenAuthenticationFilter();
+    }
+
+    /*
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/config/SecurityConfig.java]: Cleaned up

File: src/main/java/pl/agh/edu/master_diet/config/AppProperties.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public static final class OAuth2 {
+        private List<String> authorizedRedirectUris = new ArrayList<>();
+
+        public List<String> getAuthorizedRedirectUris() {
+            return authorizedRedirectUris;
+        }
+
+        public OAuth2 authorizedRedirectUris(List<String> authorizedRedirectUris) {
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/config/AppProperties.java]: Fixed

File: src/main/java/pl/agh/edu/master_diet/security/oauth2/HttpCookieOAuth2AuthorizationRequestRepository.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Component
+public class HttpCookieOAuth2AuthorizationRequestRepository
+        implements AuthorizationRequestRepository<OAuth2AuthorizationRequest> {
+
+    public static final String OAUTH2_AUTHORIZATION_REQUEST_COOKIE_NAME = "oauth2_auth_request";
+    public static final String REDIRECT_URI_PARAM_COOKIE_NAME = "redirect_uri";
+    private static final int cookieExpireSeconds = 180;
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/security/oauth2/HttpCookieOAuth2AuthorizationRequestRepository.java:18]: It'll be done later

File: src/main/java/pl/agh/edu/master_diet/security/oauth2/OAuth2AuthenticationSuccessHandler.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        httpCookieOAuth2AuthorizationRequestRepository.removeAuthorizationRequestCookies(request, response);
+    }
+
+    private boolean isAuthorizedRedirectUri(String uri) {
+        URI clientRedirectUri = URI.create(uri);
+
+        return appProperties.getOauth2().getAuthorizedRedirectUris()
+                .stream()
+                .anyMatch(authorizedRedirectUri -> {
+                    // Only validate host and port. Let the clients use different paths if they want to
```
[the remark this answers is not in the document]
Comment (jakubowiczish) [reply src/main/java/pl/agh/edu/master_diet/security/oauth2/OAuth2AuthenticationSuccessHandler.java]: Fixed

File: src/main/java/pl/agh/edu/master_diet/config/WebMvcConfig.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (RafisSomeone) [reply src/main/java/pl/agh/edu/master_diet/config/WebMvcConfig.java:10]: Ok

[the remark this answers is not in the document]
Comment (RafisSomeone) [reply src/main/java/pl/agh/edu/master_diet/config/WebMvcConfig.java:10]: Ok

File: src/main/java/pl/agh/edu/master_diet/security/oauth2/HttpCookieOAuth2AuthorizationRequestRepository.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (RafisSomeone) [reply src/main/java/pl/agh/edu/master_diet/security/oauth2/HttpCookieOAuth2AuthorizationRequestRepository.java:18]: Ok

**Pull request merged**